moved/src/state_actor.rs

```diff
@@ -225,11 +225,19 @@ impl<
 
     fn execute_transactions(
         &mut self,
-        transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
+        mut transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
         base_fee: U256,
     ) -> ExecutionOutcome {
         let mut total_tip = U256::ZERO;
         let mut outcomes = Vec::new();
+        let first_tx = transactions.next();
```

Comment:
nit: I'd leave that comment where the first transaction is always a `DepositTx`.

Reply:
I don't know the spec to reference if there is one as I got it from op-geth implementation.

Reply:
This is the only spec I see that refers to first transaction being a deposit txn.
https://github.com/ethereum-optimism/optimistic-specs/blob/main/specs/deposits.md#kinds-of-deposited-transactions

We can just simply mention that based on the op-geth implementation the first transaction must be a deposit transaction.

Other than that I think this PR is good to merge

Reply:
Great find! Also, that repository is archived and deprecated, the specs have moved to https://github.com/ethereum-optimism/specs/